modules/simplytest_submission/src/SubmissionService.php

```diff
@@ -0,0 +1,494 @@
+   */
+  protected $logger;
+
+  /**
+   * The devel dumper manager.
+   *
+   * @var \Drupal\devel\DevelDumperManagerInterface
+   */
+  protected $dumper;
+
+  /**
+   * The entity for which the script is being built.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  protected $buildEntity;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(LoggerInterface $logger, Settings $settings, DevelDumperManagerInterface $dumper = NULL) {
+    $this->logger = $logger;
+    $this->dumper = $dumper;
+
+    if (!empty($settings->get('simplytest_submission'))) {
+      $this->settings = $settings->get('simplytest_submission');
+    }
+    elseif (!empty($token = \Drupal::state()->get('simplytest_submission.service_token'))) {
+      $this->settings['service_token'] = $token;
+    }
+    else {
+      $this->logger->error('No settings found for simplytest_submission');
+    }
+  }
+
+  /**
+   * Deploys submission instance on service server.
+   *
+   * @param SubmissionInterface $entity
+   *   Entity being used to make request.
+   * @param string $script
+   *   Bash script to be used in deployment.
+   *
+   * @return array
+   *   Data response from service server.
+   *
+   * @throws \Exception
+   *   Throws an exception if server returns error.
+   */
+  public function buildInstance(SubmissionInterface $entity, $script) {
+    // Request preparation.
+    $url = SubmissionInterface::SERVICE_URL;
+    $url .= '/?' . http_build_query([
+      // Use $ drush sset simplytest_submission.service_token TOKEN.
+      // @todo provide config interface for this / or do it within rules UI
+      'token' => $this->settings['service_token'],
+      'ttl' => $entity->instance_runtime->value,
+      'image' => $entity->instance_image->value,
+      'cache' => ($entity->instance_snapshot_cache->value) ? 'true' : 'false',
+    ]);
+
+    // Send request.
+    $client = \Drupal::httpClient();
+    $response = $client->request('POST', $url, [
+      'headers' => [
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/octet-stream; charset=utf-8',
+      ],
+      'body' => $script,
+    ]);
+
+    // Check response.
+    if ($response->getStatusCode() !== 201) {
+      throw new \Exception('Invalid status code returned from spawn.sh api');
+    }
+
+    $data = json_decode($response->getBody(), TRUE);
+
+    if (!isset($data['id'])) {
+      throw new \Exception('No id in spawn.sh response data');
+    }
+
+    return $data;
+  }
+
+  /**
+   * Builds a deploy script for a submission.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity that the script is being build for.
+   *
+   * @return bool|array
+   *   Deploy script for building submission instance.
+   */
+  public function buildScript(EntityInterface $entity) {
+    $this->buildEntity = $entity;
+    $this->debug('No results from spawn.sh.');
+    $this->debug($entity, 'Entity object.');
+
```

Comment:
Good catch!! I made a note of the format of the debug method to remind myself and forgot to remove it :P 

Will remove.